src/Nova/Role.php

```diff
@@ -115,15 +132,25 @@ public static function singularLabel()
     }
 
     /**
-     * Load all permissions
+     * Load all permissions and cache for 1 minute.
+     * Enough to avoid N+1 at the Role index page,
+     * and not long enough to have them stalled.
      *
      * @return \Illuminate\Database\Eloquent\Collection
      */
     protected function loadPermissions()
     {
-        /** @var class-string */
-        $permissionClass = config('permission.models.permission');
-
-        return $permissionClass::all()->unique('name');
+        $ttl = config('permission.caching.all_permissions_ttl', now()->addMinute());
```

Comment:
Rename from `permission.cache.nova_expiration_time` to keep same config session and pattern

https://github.com/spatie/laravel-permission/blob/main/config/permission.php#L163

You can use the same config to loaa users count

Reply:
Done the config path adjustment.

Not sure about the load users.